test/controllers/coherence/session_controller_test.exs

```diff
@@ -0,0 +1,107 @@
+defmodule Ask.Coherence.SessionControllerTest do
+  use Ask.ConnCase
+  use Ask.MockGuissoCase
+
+  alias Ask.User
+
+  test "new with guisso enabled", %{conn: conn} do
+    enable_guisso()
+
+    conn = get conn, session_path(conn, :new)
+    location = redirected_to(conn, 302)
+    assert location =~ ~r{^http://guisso.localhost/oauth2/authorize}
+
+    params = URI.parse(location).query |> URI.query_decoder() |> Map.new()
+    assert params["redirect_uri"] == "http://app.ask.dev/session/oauth_callback"
+    assert params["response_type"] == "code"
+  end
+
+  test "new with guisso disabled", %{conn: conn} do
+    disable_guisso()
+
+    conn = get conn, session_path(conn, :new)
+    assert html_response(conn, 200)
+  end
+
+  describe "oauth_callback" do
+    # setup do
+    #   enable_guisso()
+    # end
```

Comment:
Is this commented for any specific reason? If its not being executed I would just delete it. What do you think?